### DISCUSSION TITLE
Should `getValues` return all form values if shouldUnregister: true?

### DISCUSSION THREAD

**Discussion by HexSleeves** [react-hook-form/react-hook-form#7832] [Ideas] [answered]
If I set `shouldUnregister = true`, set defaultValues, and call `getValues()` I still get all form values, event if that field is not registered. 

Could there be a flag passed to getValues to allow only registered fields? 
Or should getValues only return registered fields?

Curious other thoughts on this matter

**Comment by Moshyfawn** [comment, marked as answer]
I'd say, don't provide the default values, then?.. I don't really know your use-case, so it's just the first thing that comes to mind and how I handle it

**Reply by HexSleeves** [reply]
Definitely an option. I guess I was taking the lazy way out with setting defaultValues at the top level instead of at the component level per field.

**Reply by Moshyfawn** [reply]
My thought doesn't exclude setting `defaultValues`, tho. It's more along the lines of if the `defaultValue` for the input is provided, it's probably registered. That way, it's your responsibilty to provide the default value if there has to be one, as RHF just takes `defaultValues` as sets them before it can possibly know what's registered / what's not. When you start to hide, unregister etc. your inputs, that's where RHF "removes" / unregisters the values from the values state

**Reply by HexSleeves** [reply]
We are passing in an object that contains default values and some values that are used in api calls. 

To briefly explain the use case, we built a form builder where you pass it a config and it generates a form based on that config. We use react hook form to handle subscriptions and all the other nice things.

Since we don't know what the form will be until they pass in the config, it's impossible to know what to pluck from the defaultValues object so we are using the entire object.

I think in this case, I shouldn't set default values, and instead should just set the default value at the field level when I register the field
